Update samples based on GA API

changjian-wang opened the pull request:
This pull request makes significant improvements to the analyzer training sample, focusing on enhancing usability, robustness, and clarity. The main changes include a complete rewrite of the `Program.cs` workflow to provide a more interactive, user-friendly, and robust experience, improved configuration handling, and an updated analyzer creation interface. Additionally, package dependencies are updated for better compatibility.

**Major improvements to the analyzer training sample:**

*User experience and workflow:*
- Rewrote `Program.cs` to provide a step-by-step, interactive console workflow with clear instructions, input prompts, and error handling. The program now verifies prerequisites, guides users through environment variable setup, uploads training data, creates an analyzer from a structured schema, analyzes a sample document, and cleans up by deleting the analyzer.
- Added dynamic configuration resolution for training data paths and SAS URLs, with fallback prompts and validation to ensure all required information is present before proceeding.
- Implemented a utility method `ResolveDataFilePath` to robustly locate data files and folders across different execution environments.

*Analyzer creation and API interface:*
- Changed the analyzer creation method in `IAnalyzerTrainingService` to accept an explicit `analyzerId`, a dictionary-based analyzer definition, and return a `JsonDocument` with analyzer details, making the interface more flexible and consistent with modern usage patterns.

*Dependency and infrastructure updates:*
- Updated NuGet package references to the latest patch versions for `Microsoft.Extensions.*` libraries and added `Azure.Storage.Blobs` to support blob operations.

These changes make the sample more robust, easier to use, and better aligned with real-world workflows.

**References:**  
[[1]](diffhunk://#diff-d1f9cf8702b08a17d25ff87e0c6a64bcb953d57d33a6fa0ee908fc8d3bcb86ecL5-R362) [[2]](diffhunk://#diff-c0eda9f23263fe52117a400f841c44eb3100aa84a15b13b13370cebf0647e987L25-R42) [[3]](diffhunk://#diff-66fd4bdf8c5421dbb694a3d403dbedb1507cac9702072679b323d3fd4aa9bd3eL11-R14)

Commit:
Refactor and update Content Understanding integration

- Upgraded package dependencies across projects, including `Azure.AI.ContentUnderstanding` and `Microsoft.Extensions` libraries.
- Refactored `ContentExtractionService` to use `ContentUnderstandingClient` and the new `AnalyzeResult` model.
- Updated integration tests to align with the new API and replaced `JsonDocument` with `AnalyzeResult`.
- Introduced `AddContentUnderstandingClient` extension for streamlined DI configuration.
- Replaced `AZURE_CU_CONFIG` with `AZURE_CONTENT_UNDERSTANDING_ENDPOINT` and `AZURE_CONTENT_UNDERSTANDING_KEY` in configuration.
- Added `nuget.config` for local package source and upgraded test dependencies.
- Preserved legacy implementation in `OldContentExtractionService` for backward compatibility.
- General cleanup: removed unused namespaces, updated file paths, and improved error handling.

Commit:
Remove OldContentExtractionService and related logic

The `OldContentExtractionService` class, along with its methods, properties, and dependencies, has been removed. This includes:

- Removal of namespaces and usings (`ContentExtraction.Interfaces`, `ContentUnderstanding.Common`, etc.).
- Deletion of methods for document, audio, video, and video-with-face analysis (`AnalyzeDocumentAsync`, `AnalyzeAudioAsync`, `AnalyzeVideoAsync`, `AnalyzeVideoWithFaceAsync`).
- Removal of keyframe and face processing methods (`SaveKeyFrames`, `SaveFacesAndKeyFrames`).
- Elimination of supporting logic for JSON handling, image saving, and logging.

This change reflects a significant refactor or deprecation of the old content extraction functionality, likely replaced or restructured.

Commit:
Refactor and enhance Content Understanding integration

- Integrated Azure.AI.ContentUnderstanding SDK with updated package version.
- Refactored `CreateAnalyzerAsync` and `AnalyzeDocumentWithCustomAnalyzerAsync` to use `ContentAnalyzer` and `AnalyzeResult`.
- Added `BlobFileConstants` and `BlobUploadExtensions` for file handling and blob operations.
- Introduced `AnalyzeVideoWithFaceAsync` for video analysis with face detection.
- Enhanced error handling and logging for Azure service requests.
- Updated integration tests to validate `AnalyzeResult` and align with new APIs.
- Fixed potential `KeyNotFoundException` in analyzer creation and video analysis.
- Improved code readability, maintainability, and added detailed documentation.

Commit:
Enhance content extraction and field analysis

Updated `Program.cs` in `ContentExtraction` to set UTF-8 encoding and enable video face detection. Enhanced `ContentExtractionService.cs` with keyframe extraction and file extension determination. Updated `ContentUnderstanding.Common.csproj` to use a newer local package version. Refactored `IFieldExtractionService.cs` to use `AnalyzeResult`. Refactored `FieldExtraction` namespace to use `ContentAnalyzer` objects, improving analyzer setup. Switched to `ContentUnderstandingClient` in `FieldExtractionService.cs` and updated result processing. Commented out obsolete methods in `FieldExtractionService.cs`.

Commit:
Refactor for Azure AI Content Understanding

Updated namespaces to `Azure.AI.ContentUnderstanding` for better alignment with Azure services. Simplified service configuration by removing manual loading from environment variables and `appsettings.json`. Refactored analyzer creation using `ContentAnalyzer` and `FieldSchema` for improved flexibility. Enhanced error handling and assertions in tests to ensure accurate validation. Removed redundant code to reduce complexity. Updated client to `ContentUnderstandingClient` reflecting Azure SDK changes. Improved program logic with new `ContentAnalyzer` and `AnalyzeResult` types for better API alignment and functionality.

Commit:
Refactor and integrate Azure AI Content Understanding

Updated namespaces and using directives to include Azure.AI.ContentUnderstanding. Simplified service configuration in Program.cs by using AddContentUnderstandingClient. Refactored test files to use new Azure AI methods and classes, replacing file-based schema loading with in-code definitions. Consolidated classifier methods into a single ClassifyDocumentAsync method, improving error handling and logging. Updated package version in ContentUnderstanding.Common.csproj. General refactoring for improved readability, maintainability, and test coverage.

Commit:
Refactor analyzer management for Azure SDK alignment

Refactored `IManagementService` to use `ContentAnalyzer` for improved type safety and added `UpdateAnalyzerAsync` for updating analyzer properties. Simplified `DeleteAnalyzerAsync` return type.

Replaced JSON templates with programmatic `ContentAnalyzer` configurations in `Program.cs`, adding support for locales, enumerations, and detailed field definitions.

Updated `ManagementService` to use `ContentUnderstandingClient`, aligning with modern Azure SDK patterns. Enhanced logging, error handling, and detailed output for operations like creating, listing, retrieving, updating, and deleting analyzers.

Removed legacy code and redundant configurations, improving maintainability and user experience.

Commit:
Update .gitignore and clean up appsettings.json

Improved the `.gitignore` file by adding patterns to ignore:
- All `.cache` directories recursively.
- All `appsettings.*.json` files, except `appsettings.example.json`.
- Explicitly ignored `/ContentUnderstanding.Common/appsettings.json`.

Removed sensitive and environment-specific configurations from `appsettings.json`:
- Deleted `AZURE_CONTENT_UNDERSTANDING_ENDPOINT` and `AZURE_CONTENT_UNDERSTANDING_KEY`.
- Removed the `Logging` section, including log level settings.

These changes enhance security by avoiding the inclusion of sensitive data in version control and streamline the repository by ignoring unnecessary files.

Commit:
Refactor and enhance Azure Content Understanding client

- Replaced `Azure.AI.ContentUnderstanding` with a custom `AzureContentUnderstandingClient` for better control over API interactions.
- Centralized dependency management by moving `Microsoft.Extensions.Hosting` and `Microsoft.Extensions.Http` to `ContentUnderstanding.Common`.
- Updated `IContentExtractionService` to use `JsonDocument` for flexibility and added `AnalyzeDocumentFromUrlAsync`.
- Refactored `Program.cs` to use `ContentUnderstandingBootstrapper` for streamlined initialization and improved error handling.
- Introduced `ContentUnderstandingBootstrapper` and `ModelDeploymentConfiguration` to simplify setup and manage model deployments.
- Enhanced `ServiceCollectionExtensions` to support both API key and Azure AD token authentication.
- Added `SampleHelper` for JSON handling and file I/O utilities.
- Improved logging, error handling, and documentation across the codebase.
- Updated `appsettings.example.json` with placeholders for Azure AI configurations.

Commit:
Refactor analyzer service and JSON handling

Refactored `IManagementService` to use JSON-based methods, replacing object-based models. Updated `CreateAnalyzerAsync`, `ListAnalyzersAsync`, and `GetAnalyzerDetailsAsync` to work with JSON strings and `JsonElement`. Removed `UpdateAnalyzerAsync` due to deprecation.

Replaced `ContentUnderstandingClient` with `AzureContentUnderstandingClient` in `ManagementService`. Simplified analyzer creation using dictionary-based templates serialized to JSON. Improved error handling and logging.

Removed legacy code, including detailed logging and exception handling for specific cases. Introduced `System.Text.Json` for serialization with better readability and flexibility. Updated `Program.cs` to use a custom bootstrapper for service initialization.

yungshinlintw committed:
Merge remote-tracking branch 'origin/main' into changjian-wang/integrate-azure-sdk-to-samples

yungshinlintw committed:
Improve README and and update to ContentExtraction sample.

- Updated README.md to reflect the GA status of the Azure AI Content Understanding service and provide links to relevant documentation and samples.
- Added a new Content Extraction sample demonstrating the use of the Azure AI Content Understanding API for extracting semantic content from documents, audio, and video.
- Enhanced .gitignore to include the `sample_output/` directory to prevent output files from being tracked.
- Updated paths in the Content Extraction sample to simplify file handling and ensure compatibility across different execution contexts.
- Improved appsettings configuration to prioritize values from `appsettings.json` over environment variables for better usability.

yungshinlintw committed:
Enhance .gitignore to include test_output directory

- Added `**/test_output/` to the .gitignore file to prevent tracking of test output files, improving repository cleanliness and focus on source files.

yungshinlintw committed:
Add ModelDeploymentSetup project and update README for prebuilt analyzers

- Introduced a new `ModelDeploymentSetup` project to configure model deployments required for prebuilt analyzers in Azure AI Content Understanding.
- Updated `README.md` to emphasize the importance of running the `ModelDeploymentSetup` sample before using prebuilt analyzers, including detailed instructions and warnings.
- Enhanced `ContentExtraction` sample to remind users to configure model deployments before execution.
- Improved documentation across various README files to clarify setup steps and prerequisites for using the Azure AI service.

yungshinlintw committed:
Remove FieldExtractionProMode sample

yungshinlintw committed:
Remove ConversationalFieldExtraction sample project and its associated files

yungshinlintw committed:
Remove BuildPersonDirectory sample project and its associated files

yungshinlintw committed:
Clean up output

yungshinlintw committed:
Refactor FieldExtraction sample to enhance functionality and documentation

- Updated `FieldExtraction.csproj` to include `appsettings.json` and `appsettings.example.json` in the output directory for better configuration management.
- Modified `Program.cs` to improve error handling and user prompts regarding model deployment configuration, ensuring users are aware of necessary setup steps before running the sample.
- Enhanced `IFieldExtractionService` interface and `FieldExtractionService` class to streamline the analysis process with prebuilt analyzers and custom analyzers, improving clarity and usability.
- Added comprehensive `README.md` for the Field Extraction sample, detailing usage, prerequisites, key concepts, and code structure, aligning with the established documentation standards.

yungshinlintw committed:
Add e Analyzer Training sample

- Included `Azure.Storage.Blobs` package in `AnalyzerTraining.csproj` to facilitate blob storage operations.
- Refactored `Program.cs` to improve service configuration and user prompts, ensuring clarity on model deployment requirements.
- Updated `IAnalyzerTrainingService` interface and `AnalyzerTrainingService` class to streamline the creation and usage of analyzers with labeled training data.
- Created comprehensive `README.md` for the Analyzer Training sample, detailing setup, key concepts, and usage instructions, aligning with documentation standards.

yungshinlintw committed:
Implement Management Sample for Azure AI Content Understanding

- Added `Management/Program.cs` to demonstrate managing analyzers, including updating defaults, creating, listing, retrieving, and deleting analyzers.
- Introduced `Management/README.md` to provide comprehensive documentation on sample usage, prerequisites, key concepts, and code structure.
- Updated `IManagementService` interface and `ManagementService` class to support new functionalities for updating and retrieving default model deployment mappings.
- Enhanced user prompts and output messages for better clarity during sample execution, ensuring users are informed about model deployment configurations.

yungshinlintw committed:
Enhance Classifier Sample and Update Documentation

- Added a comprehensive `README.md` for the Classifier sample, detailing its capabilities, prerequisites, key concepts, code structure, and running instructions.
- Updated `Program.cs` to include validation for endpoint configuration and improved user prompts regarding model deployment setup.
- Refactored `ClassifierService` to utilize a thin REST client for document classification and analyzer creation, ensuring alignment with the Azure AI Content Understanding API.
- Included necessary appsettings files in the project configuration for better management of service settings.
- Improved error handling and output messages for clarity during sample execution, guiding users through the setup process effectively.

yungshinlintw committed:
Update README.md to Enhance Sample Descriptions

- Removed outdated entries for the `ConversationalFieldExtraction` and `BuildPersonDirectory` samples from the README.
- Updated the description for the `Management` sample to clarify its functionalities, including creating, listing, retrieving, and deleting analyzers in the Azure AI Content Understanding resource.
- Ensured consistency in documentation style and improved clarity for users navigating the sample capabilities.

yungshinlintw committed:
Update README.md to Improve Sample Descriptions and Clarity

Commit:
Remove AzureContentUnderstandingFaceClient.cs

The `AzureContentUnderstandingFaceClient.cs` file has been entirely removed from the codebase. This includes the deletion of the `AzureContentUnderstandingFaceClient` class, all its public methods (e.g., `CreatePersonDirectoryAsync`, `AddPersonAsync`, `DetectFacesAsync`), private helper methods (e.g., `CreateRequestAsync`, `SendRequestAsync`), and associated functionality.

The removed class previously provided methods for managing person directories, persons, and faces, as well as interacting with Azure Content Understanding services. This change also eliminates dependencies on `HttpClient`, `IOptions<ContentUnderstandingOptions>`, and the token provider.

This removal indicates a significant refactor or deprecation of the Azure Content Understanding client functionality.

yungshinlintw commented on .gitignore:
```diff
@@ -20,4 +20,10 @@ TestResults/
+appsettings.*.json
+!appsettings.example.json
+/ContentUnderstanding.Common/appsettings.json
+/ContentUnderstanding.Common/appsettings.json
```
Dupe

yungshinlintw commented on AnalyzerTraining/Program.cs:
```diff
@@ -2,96 +2,363 @@
             Console.WriteLine("> #################################################################################");
             Console.WriteLine(">");
-            Console.WriteLine("> Note: Currently this feature is only available for analyzer scenario is `document`");
+            Console.WriteLine("> Note: Currently, this feature is only available when the analyzer scenario is set to `document`.");
```
It should be "when baseAnalyzerId is prebuilt-document"

yungshinlintw committed:
PR changes on analyzer training

yungshinlintw committed:
Merge branch 'changjian-wang/integrate-azure-sdk-to-samples' of https://github.com/Azure-Samples/azure-ai-content-understanding-dotnet into changjian-wang/integrate-azure-sdk-to-samples

Commit:
Refactor tests and remove unused projects

- Removed unused projects: BuildPersonDirectory, FieldExtractionProMode, and ConversationalFieldExtraction, along with their references in `.csproj` and solution files.
- Refactored dependency injection to use `ContentUnderstandingBootstrapper.CreateHost()` for cleaner service setup.
- Updated integration tests:
  - Enhanced workflows and validations for `AnalyzerTraining`, `Classifier`, `ContentExtraction`, `FieldExtraction`, and `Management` tests.
  - Added new tests for document analysis from URLs and audio analysis.
  - Introduced programmatic analyzer schema definitions and scenario-specific validations.
- Removed legacy code and tests for deleted projects.
- Simplified `Program.cs` files by removing unused `using` directives and redundant host creation logic.
- Improved logging and validation for better traceability and debugging.
- Updated solution configuration to exclude deleted projects.
- Standardized JSON handling and improved code readability.

Commit:
Merge branch 'changjian-wang/integrate-azure-sdk-to-samples' of https://github.com/Azure-Samples/azure-ai-content-understanding-dotnet into changjian-wang/integrate-azure-sdk-to-samples

yungshinlintw committed:
Enhance README.md and Configuration for User Agent Tracking

- Updated `README.md` to include details about the new `AZURE_AI_USER_AGENT` configuration option, explaining its purpose, default value, and how to customize it for tracking sample usage.
- Modified `appsettings.example.json` to include the `AZURE_AI_USER_AGENT` key with a default value of `null`.
- Refactored `AzureContentUnderstandingClient.cs` to conditionally add the user agent header only if it is not null or empty.
- Enhanced `ServiceCollectionExtensions.cs` to read the user agent from configuration or environment variables, providing a default if not set.

These changes improve the clarity of user agent tracking and its configuration in the sample application.

Commit:
Merge branch 'changjian-wang/integrate-azure-sdk-to-samples' of https://github.com/Azure-Samples/azure-ai-content-understanding-dotnet into changjian-wang/integrate-azure-sdk-to-samples
